core/src/main/java/org/mapfish/print/map/geotools/grid/GridLabelFormat.java

```diff
@@ -0,0 +1,84 @@
+package org.mapfish.print.map.geotools.grid;
+
+import java.text.DecimalFormat;
+import java.text.DecimalFormatSymbols;
+
+/**
+ * Formats grid labels.
+ */
+public final class GridLabelFormat {
+
+    private String labelFormat = null;
+
+    private String valueFormat;
+    private String unitFormat;
+    private String formatDecimalSeparator;
+    private String formatGroupingSeparator;
+
+    /**
+     * Constructor.
+     * @param labelFormat Label format.
+     */
+    public GridLabelFormat(final String labelFormat) {
+        this.labelFormat = labelFormat;
+    }
+
+    /**
+     * Constructor.
+     * @param valueFormat Value format.
+     * @param unitFormat Unit format.
+     * @param formatDecimalSeparator Decimal separator.
+     * @param formatGroupingSeparator Grouping separator.
+     */
+    public GridLabelFormat(final String valueFormat, final String unitFormat,
+            final String formatDecimalSeparator, final String formatGroupingSeparator) {
+        this.valueFormat = valueFormat;
+        this.unitFormat = (unitFormat == null) ? GridParam.DEFAULT_UNIT_FORMAT : unitFormat;
+        this.formatDecimalSeparator = formatDecimalSeparator;
+        this.formatGroupingSeparator = formatGroupingSeparator;
+    }
+
+    /**
+     * Create an instance from the given config.
+     * @param param Grid param from the request.
+     */
+    public static GridLabelFormat fromConfig(final GridParam param) {
```

Comment:
Would be cleaner to have two sub classes to handle the different ways of formatting.